lib/std/Build.zig

```diff
@@ -2085,8 +2088,49 @@ pub fn run(b: *Build, argv: []const []const u8) []u8 {
     };
 }
 
+pub const SearchMethod = union(enum) {
+    /// Try to find `bin`, `lib`, and `include` sub-dirs.
+    prefix: LazyPath,
+    /// Use as written, without guessing sub-dirs.
+    /// If some path is not passed, it will be skipped.
+    paths: Paths,
+
+    pub const Paths = struct {
+        binaries: ?LazyPath = null,
+        libraries: ?LazyPath = null,
+        includes: ?LazyPath = null,
+    };
+
+    pub fn binaries(self: SearchMethod, b: *std.Build) ?LazyPath {
```

Comment:
Maybe LazyPath is not the most suitable choice here? They are used by `findProgram` in configure phase, and so can't have `generated` LazyPath.